### PR TITLE
ipmitool: depends_on x86 arch

### PR DESCRIPTION
**Pull request by chenrui333**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

ipmitool does not build arm arch
https://github.com/Homebrew/homebrew-core/actions/runs/3416649799/jobs/5686995078

relates to #113717

### CODE REVIEW

**Review by dawidd6 — APPROVED** [review]

File: Formula/ipmitool.rb [hunk 1]
```diff
@@ -16,6 +16,7 @@ class Ipmitool < Formula
   depends_on "autoconf" => :build
   depends_on "automake" => :build
   depends_on "libtool" => :build
+  depends_on arch: :x86_64
```
Comment (fxcoudert) [inline Formula/ipmitool.rb:19]: It has an `arm64_big_sur` bottle, so it's probably a different issue

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/ipmitool.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/ipmitool.rb:19]: I think this is weird case, the tool is not supposed to support x86 arch.

Comment (chenrui333) [reply Formula/ipmitool.rb:19]: We might need a new rebuild to double confirm if it is still buildable for `arm64_big_sur`

Comment (Bo98) [reply Formula/ipmitool.rb:19]: The issue was a change made in newer Xcode that only affected arm64.

The fix is probably to stop the `#pragma pack(1)` bleeding everywhere (i.e. make it `pack(push,1)` and then `pack(pop)` at the end of the header).

Comment (chenrui333) [reply Formula/ipmitool.rb:19]: > The fix is swap the order of vFlashstr, if possible. With the #pragma pack(1), the pointer is starting on byte 2.

Cool, I can maybe patch this in the upstream.

**Comment by fxcoudert** [comment]
Fixed proposed at https://github.com/Homebrew/homebrew-core/pull/117205

**Pull request closed without merging**